subsys/net/ip/net_context.c

```diff
@@ -108,6 +117,129 @@ static struct sockaddr *create_sockaddr(struct net_pkt *pkt,
 
 	return addr;
 }
+
+static int tcp_backlog_find(struct net_pkt *pkt, int *empty_slot)
+{
+	int i, empty = -1;
+
+	for (i = 0; i < CONFIG_NET_TCP_BACKLOG_SIZE; i++) {
+		if (tcp_backlog[i].tcp == NULL && empty < 0) {
+			empty = i;
+			continue;
+		}
+
+		if (net_pkt_family(pkt) != tcp_backlog[i].remote.family)
+			continue;
+
+		switch (net_pkt_family(pkt)) {
+#if defined(CONFIG_NET_IPV6)
+		case AF_INET6:
+			if (net_sin6(&tcp_backlog[i].remote)->sin6_port !=
+			    NET_TCP_HDR(pkt)->src_port)
+				continue;
+
+			if (memcmp(&net_sin6(&tcp_backlog[i].remote)->sin6_addr,
+				   &NET_IPV6_HDR(pkt)->src,
+				   sizeof(struct in6_addr)))
+				continue;
+
+			break;
+#endif
+#if defined(CONFIG_NET_IPV4)
+		case AF_INET:
+			if (net_sin(&tcp_backlog[i].remote)->sin_port !=
+			    NET_TCP_HDR(pkt)->src_port)
+				continue;
+
+			if (memcmp(&net_sin(&tcp_backlog[i].remote)->sin_addr,
+				   &NET_IPV4_HDR(pkt)->src,
+				   sizeof(struct in_addr)))
+				continue;
+
+			break;
+#endif
+		}
+
+		return i;
+	}
+
+	if (empty_slot)
+		*empty_slot = empty;
+
+	return -EADDRNOTAVAIL;
+}
+
+static int tcp_backlog_syn(struct net_pkt *pkt, struct net_context *context)
+{
+	int empty_slot = -1;
+
+	if (tcp_backlog_find(pkt, &empty_slot) >= 0)
+		return -EADDRINUSE;
+
+	if (empty_slot < 0)
+		return -ENOMEM;
+
+	tcp_backlog[empty_slot].tcp = context->tcp;
+
+	(void) create_sockaddr(pkt, &tcp_backlog[empty_slot].remote);
```

Comment:
I didn't see this pattern (explicit void cast of function call) to be used elsewhere in Zephyr codebase.